File: TP2-Java/Program Java/Produk.java
public class Produk {
    // Atribut menggunakan access modifier 'protected'
    protected String id_produk;
    protected String asal_produk;
    protected double harga;

    // Constructor untuk inisialisasi objek
    public Produk(String id_produk, String asal_produk, double harga) {
        this.id_produk = id_produk;
        this.asal_produk = asal_produk;
        this.harga = harga;
    }

    // Getter methods
    public String getId_produk() {
        return id_produk;
    }

    public String getAsal_produk() {
        return asal_produk;
    }

    public double getHarga() {
        return harga;
    }

}

File: TP2-CPP/Kode Program TP 2/Produk.cpp
#include <iostream>
#include <string>

class Produk {
protected:
    // Atribut menggunakan access specifier 'protected'
    std::string id_produk;
    std::string asal_produk;
    double harga;

public:
    // Constructor untuk inisialisasi objek
    Produk(std::string id_produk, std::string asal_produk, double harga) {
        this->id_produk = id_produk;
        this->asal_produk = asal_produk;
        this->harga = harga;
    }

    // Getter methods untuk mengambil data
    std::string getId_produk() const {
        return id_produk;
    }

    std::string getAsal_produk() const {
        return asal_produk;
    }

    double getHarga() const {
        return harga;
    }

};